bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -529,4 +530,15 @@ private static boolean isJpeg(byte[] data) {
                 && data[data.length - 2] == (byte) 0xFF && data[data.length - 1] == (byte) 0xD9);
     }
 
+    @Reference
+    protected void setHttpClientFactory(final HttpClientFactory httpClientFactory) {
+        HttpUtil.httpClientFactory = httpClientFactory;
+    }
+
+    protected void unsetHttpClientFactory(final HttpClientFactory httpClientFactory) {
+        if (HttpUtil.httpClientFactory == httpClientFactory) {
```

Comment:
This if check will not be necessary since the service reference is getting injected with static policy.